cache.py

```diff
@@ -1,11 +1,22 @@
 import os
 from collections import OrderedDict
 import uuid
+from dataclasses import dataclass
 
 from pytube import YouTube
 
 from urllib.parse import urlparse, parse_qs
 
+@dataclass
+class VideoInfo():
+    video_id: str
```

Comment:
we dont need this field, because in the cache, the video id will map to the dataclass, so storing the video id in both the cache key and here is redundant

lmk if this makes sense

Reply:
yes makes sense